R/utils.R

```diff
@@ -69,15 +69,31 @@ link_arms <- function(
 #' Takes a string separated by \code{,}s and/or \code{|}s (i.e. comma/tab separated values) containing key value pairs (\code{raw} and \code{label}) and returns a tidy tibble.
 #'
 #' @param string A \code{db_metadata$select_choices_or_calculations} field pre-filtered for checkbox \code{field_type}
-#' @param raw_or_label A string (either 'raw' or 'label') that specifies whether to export the raw coded values or the labels for the options of multiple choice fields.
 #'
-#' @import dplyr
+#' @import dplyr stringi
 #' @keywords internal
 
-parse_labels <- function(string, raw_or_label){
+parse_labels <- function(string){
   out <- string %>%
-    strsplit(" \\| |, ") %>% # split either by ' | ' or ', '
+    strsplit(" \\| ") # Split by "|"
+
+  # Check there is a comma in all | delimited strsplit elements
+  if (!all(grepl(",", out[[1]]))) {
+    stop(paste0("The raw/label structure of the string is not analyzable: ", string))
```

Comment:
Do you want to use `checkmate::assert` here? 

Also, I would change the message to: "Cannot parse the select_choices_or_calculations field from REDCap metadata. This may happen if there is a pipe character \`|\` inside the label"

Reply:
I tried to initially but couldn't figure out how to get a custom error message out of `assert`/`check`. The default when I used it wasn't helpful for the user., i.e. `Error: Assertion on 'all(grepl(",", out[[1]]))' failed: Must be TRUE.`

I would still like to keep the `string` output because I think it can be helpful to the user to locate what field is causing the issue. Otherwise agree, implementing now.